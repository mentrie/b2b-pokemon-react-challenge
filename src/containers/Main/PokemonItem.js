import React from 'react';
import PropTypes from 'prop-types';
import Link from 'react-router-dom/Link';
import { getPokemanId, pokemonImageUrl, formatName } from '../../helpers/utils';

const PokemonItem = ({ pokemon, onSelect }) => {
  const id = getPokemanId(pokemon.url);

  return (
    <Link to={`/${pokemon.name}/${id}`} onClick={onSelect} className="main__pokemon__item">
      <img src={pokemonImageUrl(id)} alt={pokemon.name} />
    </Link>
  );
}

PokemonItem.propTypes = {
  pokemon: PropTypes.shape({
    id: PropTypes.number,
    url: PropTypes.string
  })
}

export default PokemonItem;