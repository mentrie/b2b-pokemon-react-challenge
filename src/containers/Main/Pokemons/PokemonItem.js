import React from 'react';
import PropTypes from 'prop-types';
import { getPokemonId, pokemonImageUrl, formatName } from '../../../helpers/utils';

const PokemonItem = ({ pokemon }) => {
  const id = getPokemonId(pokemon.url);
  const backgroundStyle = {
    backgroundImage: `url(${pokemonImageUrl(id)})`
  };
  return (
    <div className="pokemon__item">
      <div className="pokemon__item__background__cover" style={backgroundStyle} />
      <div className="pokemon__item__content">
        <img className="pokeman__item__picture" src={pokemonImageUrl(id)} alt={`pokemon-${id}`}/>
        <div className="pokemon__item__name">{formatName(pokemon.name)}</div>
      </div>
    </div>
  );
};

PokemonItem.propTypes = {
  pokemon: PropTypes.shape({
    id: PropTypes.number,
    url: PropTypes.string
  })
};

export default PokemonItem;
