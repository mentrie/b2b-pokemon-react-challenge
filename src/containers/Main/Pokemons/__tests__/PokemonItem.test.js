import React from 'react';
import PokemonItem from '../PokemonItem';

test('render component', () => {
  const props = {
    pokemon: {
      name: 'test',
      url: 'test/2'
    }
  };

  const component = shallow(<PokemonItem {...props} />);
  expect(component).toMatchSnapshot();
});
