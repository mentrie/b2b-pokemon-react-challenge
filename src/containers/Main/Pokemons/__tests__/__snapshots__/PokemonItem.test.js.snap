// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component 1`] = `
<div
  className="pokemon__item"
>
  <div
    className="pokemon__item__background__cover"
    style={
      Object {
        "backgroundImage": "url(https:raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/NaN.png)",
      }
    }
  />
  <div
    className="pokemon__item__content"
  >
    <div
      className="pokeman__item__picture"
      style={
        Object {
          "backgroundImage": "url(https:raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/NaN.png)",
        }
      }
    />
    <div
      className="pokemon__item__name"
    >
      Test
    </div>
  </div>
</div>
`;
